base/distributed/pmap.jl

```diff
@@ -293,3 +306,18 @@ function batchsplit(c; min_batch_count=1, max_batch_size=100)
 
     return Iterators.flatten((head, tail))
 end
+
+function isanon_function(f)
+    t = typeof(f)
+    tn = t.name
+    if isdefined(tn, :mt)
+        name = tn.mt.name
+        mod = tn.module
+        if t.super === Function &&
+            unsafe_load(Base.unsafe_convert(Ptr{UInt8}, tn.name)) == UInt8('#') &&
+            (!isdefined(mod, name) || t != typeof(getfield(mod, name)))
+            return true
+        end
+    end
+    return false
+end
```

Comment:
@vtjnash / @JeffBezanson , this is more or less an adaptation of the code in https://github.com/JuliaLang/julia/blob/bf47fd90f50bfda87fad5a501829e91a60c7d0d0/base/serialize.jl#L500-L517 without a deeper understanding of how this is supposed to work. Can you review this for correctness? Or suggest the correct way to test if a function is an anonymous function?